### DISCUSSION TITLE
Can I create an api key that does not expire?  If not, how can I create one on the fly?

### DISCUSSION THREAD

**Discussion by cloudlady911** [drakkan/sftpgo#1472] [General]
I am trying to develop a bash script that will work in my CD system (Octopus Deploy).  This script will create an account in sftpgo for a user and connect the account to their S3 Bucket which is created in a previous step.

I don't really want to have to get a new API key every time this script is run.  But if I have to, then I need to know how to do this with a curl script.  Also, I don't want to expose my own credentials.

I have tried testing with the Swagger API but continuously get invalid token error.

**Comment by cloudlady911** [comment]
I figured out the answer myself.  The process is to get the token then use the token to create the API key

**Reply by zhuyue0825** [reply]
> I figured out the answer myself. The process is to get the token then use the token to create the API key
 how can i use the token to create the API key
